### ISSUE TITLE
Can't map 'format'

### ISSUE THREAD

**Issue by gwhitney** [josdejong/mathjs#2436]
From the demo on the front page:
```
map([1, 4, 5], format)

  TypeError: Unexpected type of argument in function format 
    (expected: number or function or Object or string or boolean, 
     actual: Array, index: 1)
```

**Comment by josdejong** [comment]
The function `map` in mathjs passes three arguments: the value of the element, the index of the element, and the matrix being traversed ([see docs](https://mathjs.org/docs/reference/functions/map.html)). So the function `format` gets these three arguments passed and it doesn't understand the second and third argument.

You can make it work with an intermediate, untyped function:

```
map([1, 4, 5], f(x) = format(x))
```

It would be very nice if this would "just" work though. Any ideas?

**Comment by gwhitney** [comment]
Well could the guts of map check the max arguments of its argument and if it's fewer than three, only supply as many as it will take?

**Comment by josdejong** [comment]
Such a solution makes sense. Function `format` accepts one or two arguments, so that's a bit complicated in general. To make it work, `map` has to check not only the number of arguments, but also whether the argument types match (in case of a typed-function). That could work I think.

**Comment by josdejong** [comment]
Published in `v10.3.0`
